samples/xbee/demos/EndToEndSample/main.py

```diff
@@ -58,10 +66,22 @@ def xbee_data_received_callback(msg):
 
         # Upload the temperature and humidity values to Digi Remote Manager.
         try:
-            datapoint.upload(DATA_STREAM_TEMP % src_addr, data_split[0], units="C",
-                             data_type=datapoint.DataType.DOUBLE)
-            datapoint.upload(DATA_STREAM_HUM % src_addr, data_split[1], units="%",
-                             data_type=datapoint.DataType.DOUBLE)
+            if is_old_firmware():
+                datapoint.upload(DATA_STREAM_TEMP % src_addr, data_split[0],
+                                 units="C", data_type=DataType.DOUBLE)
+                datapoint.upload(DATA_STREAM_HUM % src_addr, data_split[1],
+                                 units="%", data_type=DataType.DOUBLE)
+            else:
+                # Check if DataPoint module is imported.
+                if "DataPoint" not in sys.modules:
```

Comment:
I would not check this if this is implied in the firmware used. But if we want to this verification, I would throw and exception or print a message telling this support is not available and do not continue.

Reply:
I think the implementation is correct. The firmware check dtermines whether the upload multiple and DataPoint module can be used. The import check is done to avoid re-importing the module with everytime the method is called, it is not to check whether we can use or not the module.

Reply:
I do not think we need that kind of check then. The module will only be imported once although you call several times to the import.

Reply:
Ok, will remove the check then.